x-pack/plugins/reporting/server/routes/generate/csv_searchsource_immediate.ts

```diff
@@ -69,58 +68,70 @@ export function registerGenerateCsvFromSavedObjectImmediate(
         const logger = parentLogger.clone([CSV_SEARCHSOURCE_IMMEDIATE_TYPE]);
         const runTaskFn = runTaskFnFactory(reporting, logger);
         const requestHandler = new RequestHandler(reporting, user, context, req, res, logger);
-
         const eventLog = reporting.getEventLogger({
           jobtype: CSV_SEARCHSOURCE_IMMEDIATE_TYPE,
           created_by: user && user.username,
           payload: { browserTimezone: (req.params as BaseParams).browserTimezone },
         });
 
-        eventLog.logExecutionStart();
+        const stream = new (class extends PassThrough {
+          private onFirstByte?(): void;
```

Comment:
I think it will be best to define this class in a separate file, perhaps `x-pack/plugins/reporting/server/routes/lib/stream_passthrough.ts`

Reply:
Thanks for pointing this out. I've also covered that with unit tests.